src/aspire/noise/noise.py

```diff
@@ -345,3 +354,264 @@ def estimate_noise_psd(self):
         noise_psd_est[mid, mid] -= mean_est**2
 
         return noise_psd_est
+
+
+class LegacyNoiseEstimator(NoiseEstimator):
+    """
+    Isotropic noise estimator ported from MATLAB `cryo_noise_estimation`.
+    """
+
+    def __init__(self, src, bg_radius=None, max_d=None):
+        """
+        Given an `ImageSource`, constructs
+
+        :param src: A `ImageSource` object.
+        :param bg_radius: The radius of the disk whose complement is used to estimate the noise.
+            Radius is relative proportion, where `1` represents
+            the radius of disc inscribing a `(src.L, src.L)` image.
+            Default of `None` uses `(np.floor(src.L / 2) - 1) / L`
+        :param max_d: Max computed correlation distance as a proportion of `src.L`.
+            Default of `None` uses `np.floor(src.L/3) / L`.
+        """
+
+        if bg_radius is None:
+            bg_radius = (np.floor(src.L / 2) - 1) / src.L
+
+        super().__init__(src=src, bg_radius=bg_radius, batch_size=1)
+
+        self.max_d = max_d
+        if self.max_d is None:
+            self.max_d = np.floor(src.L / 3) / src.L
+
+    def estimate(self):
+        """
+        :return: The estimated noise variance of the images.
+        """
+
+        return np.mean(self.filter.evaluate_grid(self.src.L))
+
+    def _create_filter(self):
+        """
+        :return: The estimated noise power spectral distribution (PSD) of the images in the form of a filter object.
+        """
+        return ArrayFilter(self._estimate_noise_psd())
+
+    def _estimate_noise_psd(self):
+        """
+        :return: The estimated noise variance of the images in the Source used to create this estimator.
+        """
+        # Setup parameters
+        samples_idx = grid_2d(self.src.L, normalized=False, shifted=True)["r"] >= (
+            self.bg_radius * self.src.L
+        )
+        max_d_pixels = round(self.max_d * self.src.L)
+
+        psd = self.epsdS(self.src.images, samples_idx, max_d_pixels)[0]
+
+        return psd
+
+    @staticmethod
+    def epsdR(images, samples_idx, max_d=None):
```

Comment:
Maybe we can come up with better names for `epsdR` and `epsdS` (more in line with our naming conventions) and then just mention in the docstring that these are ports of `cryo_epsdR(S)`. The `R` seems to stand for "radial", does `S` stand for "spatial"?

Reply:
Well, I left them intentionally to match the MATLAB code so I wouldn't have to explain that they match some function with a different name... but yeah they can be changed.

Reply:
Updated.

Reply:
Cool. Thanks